### PR TITLE
RUM-2624 Uniformize optional schema attributes

### PR DESCRIPTION
**Pull request by maciejburda**
### What and why?

Adds default values in accordance to strategy 1 mentioned in the rfc: https://datadoghq.atlassian.net/wiki/spaces/RUMP/pages/2638055672/RFC+-+Uniformize+optional+schema+attributes

I used this excel sheet to track and comment on described properties. There're some tricky ones to solve.
Also not sure if I should be removing those which are tagged: `Do not send`.
https://docs.google.com/spreadsheets/d/1jdUJpE_1eArzgJ1R8XOzSiziG5n6Z_k2-ftwSEkRFl8/edit#gid=0

### How?

Approach for finding the required changes was:
1. Temporarily make required field non-optional in the generated code
2. Follow the breadcrumbs of the compiler and provide default values
3. Revert changes to generated code
4. Check tests and apply proper fixes

### Review checklist
- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes

### Custom CI job configuration (optional)
- [ ] Run unit tests
- [ ] Run integration tests
- [ ] Run smoke tests

### CODE REVIEW

File: Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewScope.swift [hunk 1]
```diff
@@ -526,7 +526,7 @@ internal class RUMViewScope: RUMScope, RUMContextProvider {
             version: context.version,
             view: .init(
                 id: self.context.activeViewID.orNull.toRUMDataFormat,
-                inForeground: nil,
+                inForeground: false,
```
Comment (plousada) [inline Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewScope.swift:529]: I believe this is a browser only attribute.

File: Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewScope.swift [hunk 2]
```diff
@@ -550,7 +550,7 @@ internal class RUMViewScope: RUMScope, RUMContextProvider {
         let longTaskEvent = RUMLongTaskEvent(
             dd: .init(
                 browserSdkVersion: nil,
-                discarded: nil,
+                discarded: false,
```
Comment (plousada) [inline Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewScope.swift:553]: No value in setting this one. If false won't have any effect and would only increase payload size.

**Comment by plousada** [comment]
`view.crash`, `action.crash`, `action.error`, `action.longtask` and `action.resource` shouldn't be nil.

**Review by plousada — CHANGES_REQUESTED** [review]

File: Sources/Datadog/FeaturesIntegration/CrashReporting/CrashReportingWithRUMIntegration.swift [hunk 3]
```diff
@@ -257,7 +257,7 @@ internal struct CrashReportingWithRUMIntegration: CrashReportingIntegration {
             version: lastRUMView.version,
             view: .init(
                 id: lastRUMView.view.id,
-                inForeground: nil,
+                inForeground: false,
```
Comment (maciejburda) [inline Sources/Datadog/FeaturesIntegration/CrashReporting/CrashReportingWithRUMIntegration.swift:260]: revert?

File: Sources/Datadog/FeaturesIntegration/CrashReporting/CrashReportingWithRUMIntegration.swift [hunk 4]
```diff
@@ -368,7 +368,7 @@ internal struct CrashReportingWithRUMIntegration: CrashReportingIntegration {
             os: .init(context: context),
             service: context.service,
             session: .init(
-                hasReplay: nil,
+                hasReplay: false,
```
Comment (maciejburda) [inline Sources/Datadog/FeaturesIntegration/CrashReporting/CrashReportingWithRUMIntegration.swift:371]: revert?

File: Sources/Datadog/RUM/RUMMonitor/Scopes/RUMResourceScope.swift [hunk 5]
```diff
@@ -135,7 +135,7 @@ internal class RUMResourceScope: RUMScope {
         let resourceEvent = RUMResourceEvent(
             dd: .init(
                 browserSdkVersion: nil,
-                discarded: nil,
+                discarded: false,
```
Comment (maciejburda) [inline Sources/Datadog/RUM/RUMMonitor/Scopes/RUMResourceScope.swift:138]: revert?

File: Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewScope.swift [hunk 6]
```diff
@@ -361,7 +361,7 @@ internal class RUMViewScope: RUMScope, RUMContextProvider {
             os: .init(context: context),
             service: context.service,
             session: .init(
-                hasReplay: nil,
+                hasReplay: false,
```
Comment (maciejburda) [inline Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewScope.swift:364]: revert?

File: Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewScope.swift [hunk 7]
```diff
@@ -371,7 +371,7 @@ internal class RUMViewScope: RUMScope, RUMContextProvider {
             version: context.version,
             view: .init(
                 id: viewUUID.toRUMDataFormat,
-                inForeground: nil,
+                inForeground: false,
```
Comment (maciejburda) [inline Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewScope.swift:374]: revert?

File: Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewScope.swift [hunk 8]
```diff
@@ -415,7 +415,7 @@ internal class RUMViewScope: RUMScope, RUMContextProvider {
             os: .init(context: context),
             service: context.service,
             session: .init(
-                hasReplay: nil,
+                hasReplay: false,
```
Comment (maciejburda) [inline Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewScope.swift:418]: revert?

File: Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewScope.swift [hunk 9]
```diff
@@ -563,11 +563,11 @@ internal class RUMViewScope: RUMScope, RUMContextProvider {
             date: (command.time - command.duration).addingTimeInterval(serverTimeOffset).timeIntervalSince1970.toInt64Milliseconds,
             device: .init(context: context),
             display: nil,
-            longTask: .init(duration: taskDurationInNs, id: nil, isFrozenFrame: isFrozenFrame),
+            longTask: .init(duration: taskDurationInNs, id: "", isFrozenFrame: isFrozenFrame), // TODO
             os: .init(context: context),
             service: context.service,
             session: .init(
-                hasReplay: nil,
+                hasReplay: false,
```
Comment (maciejburda) [inline Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewScope.swift:570]: revert?

File: Sources/Datadog/RUM/RUMTelemetry.swift [hunk 10]
```diff
@@ -224,39 +224,39 @@ private extension FeaturesConfiguration {
...
+            useAllowedTracingOrigins: false,
+            useAttachToExisting: false,
+            useBeforeSend: false,
+            useCrossSiteSessionCookie: false,
+            useExcludedActivityUrls: false,
             useFirstPartyHosts: !(self.rum?.firstPartyHosts.isEmpty ?? true),
             useLocalEncryption: self.common.encryption != nil,
             useProxy: self.common.proxyConfiguration != nil,
-            useSecureSessionCookie: nil,
+            useSecureSessionCookie: false,
```
Comment (maciejburda) [inline Sources/Datadog/RUM/RUMTelemetry.swift:259]: revert?

**Review by maciejburda — COMMENTED** [review]
One additional "quick" strategy I followed was removing all the optional flags, but I think it went out of hand.

I marked some of the questionable changes that I should probably revert.

Two remaining questions I have are about not sending some fields (blue color in the sheet below) and fields that are tricky to default with the current data flow (yellow color).
Would be great to get some insights.

https://docs.google.com/spreadsheets/d/1jdUJpE_1eArzgJ1R8XOzSiziG5n6Z_k2-ftwSEkRFl8/edit#gid=0

File: Tests/DatadogTests/Datadog/RUM/RUMMonitor/Scopes/RUMViewScopeTests.swift [hunk 11]
```diff
@@ -1059,7 +1059,7 @@ class RUMViewScopeTests: XCTestCase {
         XCTAssertEqual(error.error.source, .source)
         XCTAssertEqual(error.error.sourceType, .ios)
         XCTAssertNil(error.error.stack)
-        XCTAssertNil(error.error.isCrash)
+        XCTAssertTrue(error.error.isCrash == false)
```
Comment (maxep) [inline Tests/DatadogTests/Datadog/RUM/RUMMonitor/Scopes/RUMViewScopeTests.swift:1062]: /nit
```suggestion
        XCTAssertFalse(error.error.isCrash)
```

**Review by maxep — APPROVED** [review]
LGTM 👍

**Review by plousada — APPROVED** [review]

File: Tests/DatadogTests/Datadog/RUM/RUMMonitor/Scopes/RUMViewScopeTests.swift (hunk 11 above)
Comment (maciejburda) [reply Tests/DatadogTests/Datadog/RUM/RUMMonitor/Scopes/RUMViewScopeTests.swift:1062]: Of course! I think I was frantically fighting with linter :D

**Pull request merged**